sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1326,7 +1326,11 @@ class Analyzer(
      * Note : In this routine, the unresolved attributes are resolved from the input plan's
      * children attributes.
      */
-    private def resolveExpressionTopDown(e: Expression, q: LogicalPlan): Expression = {
+    private def resolveExpressionTopDown(
+        e: Expression,
+        q: LogicalPlan,
+        trimAlias: Boolean = false,
+        isTopLevel: Boolean = false): Expression = {
```

Comment:
`isTopLevel` should be true by default, otherwise the semantic is wrong when callers call `resolveExpressionTopDown` without specifying `isTopLevel`.

Reply:
> `isTopLevel` should be true by default, otherwise the semantic is wrong when callers call `resolveExpressionTopDown` without specifying `isTopLevel`.

Got your point, make this param controlled by this method  make this method more safety.